File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        current, maximum = 0, 0
        for i in range(1, len(prices)):
            delta = prices[i] - prices[i-1]
            current += delta
            if current < 0:
                current = 0;
            maximum = max(maximum, current)
        return maximum

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
// KADANE'S ALGORITHM
const maxProfit = (prices) => {
  let max = 0;
  let current = 0;
  for (let i = 1; i < prices.length; i++) {
    let delta = prices[i] - prices[i-1];
    current += delta;
    if (current < 0) current = 0;
    max = Math.max(max, current);
  }
  return max;
};